connector/data-source-handler.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+class DataSourceHandler {
+  static createDataSource(workspace, id, data, cb) {
+    function create(next) {
+      workspace.addDataSource(id, data, function(err) {
```

Comment:
the tasks are run by async.queue , which does not provide a err object, there is only next ()